lib/parse.js

```diff
@@ -250,7 +251,20 @@ Parse.prototype._startEntry = function (c) {
 
   if (onend) entry.on("end", onend)
 
+  if (entry.type === "File") {
```

Comment:
I'm not so sure if it's customary in this project, but I'd love to see a few comments explaining how this mitigates the security issue. May be valuable for someone maintaining this code in the future. Thanks!

Reply:
Another request would be to cover the edge cases that exercise this logic in unit tests.